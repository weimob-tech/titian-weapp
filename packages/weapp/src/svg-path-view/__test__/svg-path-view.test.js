const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', async () => {
    const comp = getComponent('svg-path-view', { name: 'checkbox' });

    const clickCallback = jest.fn();
    comp.addEventListener('click', clickCallback);

    expect(comp).toMatchSnapshot();
    comp.setData({ name: 'close-double', fills: null });
    await Promise.resolve();
    expect(comp).toMatchSnapshot();

    comp.setData({ name: 'close-double', fills: ['#000', '#ccc'] });
    await Promise.resolve();
    expect(comp).toMatchSnapshot();
    comp.querySelector('.titian-svg-path-view').dispatchEvent('tap');
    await Promise.resolve();
    expect(clickCallback).toHaveBeenCalled();
    clickCallback.mockClear();

    comp.setData({ name: 'checkbox', fills: ['#000', '#ccc'] });
    await Promise.resolve();
    expect(comp).toMatchSnapshot();

    comp.setData({ name: 'checkbox', fills: 'rgba(0,0,0,1)' });
    await Promise.resolve();
    expect(comp).toMatchSnapshot();
    comp.querySelector('.titian-svg-path-view').dispatchEvent('tap');
    await Promise.resolve();
    expect(clickCallback).toHaveBeenCalled();
    clickCallback.mockClear();
  });
});
