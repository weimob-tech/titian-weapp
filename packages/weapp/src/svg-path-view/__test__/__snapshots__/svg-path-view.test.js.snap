// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="titian-svg-path-view  ext-class"
    style="background-image: url(\\"data:image/svg+xml, %3Csvg viewBox='0 0 1024 1024' xmlns='http://www.w3.org/2000/svg'  %3E%3Cpath d='M1024 0v1024H0V0h1024z m-253.216 358.656a40 40 0 0 0-53.376-1.76l-3.2 3.008-262.08 273.664-143.968-134.24-2.368-2.048a40 40 0 0 0-55.2 57.408l3.008 3.168 201.696 188.096 316.704-330.72 2.08-2.368c13.12-16 11.904-39.648-3.296-54.208z'   fill='rgb(255, 44, 25)'/%3E%3C/svg%3E\\")"
    bind:tap="onClick"
  />
</main>
`;

exports[`test component should render success 2`] = `
<main>
  <wx-view
    class="titian-svg-path-view  ext-class"
    style="background-image: url(\\"data:image/svg+xml, %3Csvg viewBox='0 0 1024 1024' xmlns='http://www.w3.org/2000/svg'  %3E%3Cpath d='M210.289778 210.289778c-166.599111 166.627556-166.599111 436.792889 0 603.420444 166.627556 166.599111 436.792889 166.599111 603.420444 0 166.599111-166.627556 166.599111-436.792889 0-603.420444-166.627556-166.599111-436.792889-166.599111-603.420444 0z'   fill='rgb(158, 158, 158)'/%3E %3Cpath d='M342.044444   631.978667l2.588445-2.901334L461.738667 512l-117.105778-117.077333a35.555556 35.555556 0 0 1 47.388444-52.878223l2.901334 2.588445L512 461.738667l117.077333-117.105778a35.555556 35.555556 0 0 1 52.878223 47.388444l-2.588445 2.901334L562.261333 512l117.105778 117.077333a35.555556 35.555556 0 0 1-47.388444 52.878223l-2.901334-2.588445L512 562.261333l-117.077333 117.105778a35.555556 35.555556 0 0 1-52.878223-47.388444z'   fill='rgb(255, 255, 255)'/%3E%3C/svg%3E\\")"
    bind:tap="onClick"
  />
</main>
`;

exports[`test component should render success 3`] = `
<main>
  <wx-view
    class="titian-svg-path-view  ext-class"
    style="background-image: url(\\"data:image/svg+xml, %3Csvg viewBox='0 0 1024 1024' xmlns='http://www.w3.org/2000/svg'  %3E%3Cpath d='M210.289778 210.289778c-166.599111 166.627556-166.599111 436.792889 0 603.420444 166.627556 166.599111 436.792889 166.599111 603.420444 0 166.599111-166.627556 166.599111-436.792889 0-603.420444-166.627556-166.599111-436.792889-166.599111-603.420444 0z'   fill='rgb(0, 0, 0)'/%3E %3Cpath d='M342.044444   631.978667l2.588445-2.901334L461.738667 512l-117.105778-117.077333a35.555556 35.555556 0 0 1 47.388444-52.878223l2.901334 2.588445L512 461.738667l117.077333-117.105778a35.555556 35.555556 0 0 1 52.878223 47.388444l-2.588445 2.901334L562.261333 512l117.105778 117.077333a35.555556 35.555556 0 0 1-47.388444 52.878223l-2.901334-2.588445L512 562.261333l-117.077333 117.105778a35.555556 35.555556 0 0 1-52.878223-47.388444z'   fill='rgb(204, 204, 204)'/%3E%3C/svg%3E\\")"
    bind:tap="onClick"
  />
</main>
`;

exports[`test component should render success 4`] = `
<main>
  <wx-view
    class="titian-svg-path-view  ext-class"
    style="background-image: url(\\"data:image/svg+xml, %3Csvg viewBox='0 0 1024 1024' xmlns='http://www.w3.org/2000/svg'  %3E%3Cpath d='M1024 0v1024H0V0h1024z m-253.216 358.656a40 40 0 0 0-53.376-1.76l-3.2 3.008-262.08 273.664-143.968-134.24-2.368-2.048a40 40 0 0 0-55.2 57.408l3.008 3.168 201.696 188.096 316.704-330.72 2.08-2.368c13.12-16 11.904-39.648-3.296-54.208z'   fill='rgb(0, 0, 0)'/%3E%3C/svg%3E\\")"
    bind:tap="onClick"
  />
</main>
`;

exports[`test component should render success 5`] = `
<main>
  <wx-view
    class="titian-svg-path-view  ext-class"
    style="background-image: url(\\"data:image/svg+xml, %3Csvg viewBox='0 0 1024 1024' xmlns='http://www.w3.org/2000/svg'  %3E%3Cpath d='M1024 0v1024H0V0h1024z m-253.216 358.656a40 40 0 0 0-53.376-1.76l-3.2 3.008-262.08 273.664-143.968-134.24-2.368-2.048a40 40 0 0 0-55.2 57.408l3.008 3.168 201.696 188.096 316.704-330.72 2.08-2.368c13.12-16 11.904-39.648-3.296-54.208z'   fill='rgba(0,0,0,1)'/%3E%3C/svg%3E\\")"
    bind:tap="onClick"
  />
</main>
`;
